tests/models/test_validator.py

```diff
@@ -0,0 +1,145 @@
+"""Tests for the event validator"""
+
+import json
+import logging
+
+import pytest
+
+from ralph.exceptions import BadFormatException, UnknownEventException
+from ralph.models.selector import ModelSelector
+from ralph.models.validator import Validator
+
+from tests.fixtures.edx.browser import PageCloseBrowserEventFactory
+from tests.fixtures.edx.server import ServerEventFactory
+
+
+@pytest.mark.parametrize("ignore_errors", [True, False])
+@pytest.mark.parametrize("fail_on_unknown", [True, False])
+def test_models_validator_validate_with_no_events(
+    ignore_errors, fail_on_unknown, caplog
+):
+    """Tests given no events, the validate method does not write error messages."""
+
+    result = Validator(ModelSelector()).validate([], ignore_errors, fail_on_unknown)
+    with caplog.at_level(logging.ERROR):
+        assert list(result) == []
+    assert [] == [message for _, _, message in caplog.record_tuples]
+
+
+@pytest.mark.parametrize("ignore_errors", [True, False])
+@pytest.mark.parametrize("fail_on_unknown", [True, False])
+def test_models_validator_validate_with_invalid_json_events(
```

Comment:
It is confusing with the docstring description. The test function name is `with_invalid_json_events` and the docstring mentions that it is invalid but not JSON. I would suggest: 
```suggestion
def test_models_validator_validate_with_invalid_events(
```
What do you think?

Reply:
Good Point) Thanks) Indeed the test name is confusing, replaced it with `test_models_validator_validate_with_non_json_events`